### PR TITLE
refactor: Remove Dataworker UBA functionality

### PR DESCRIPTION
**Pull request by pxrl**
UBA support is hampering updates for v3. Drop it for now; we can re-introduce it later once we're over the v3 hump.

### CODE REVIEW

File: src/dataworker/Dataworker.ts [hunk 1]
```diff
@@ -352,28 +346,20 @@ export class Dataworker {
...
     this.logger.debug({
       at: "Dataworker#propose",
-      message: `Proposing ${isUBA ? "UBA" : "Legacy"} root bundle`,
+      message: "Proposing root bundle",
       blockRangesForProposal,
     });
-    const rootBundleData = { ...(await rootBundleDataProducer) };
+    const rootBundleData = {
+      ...(await this._proposeRootBundle(blockRangesForProposal, spokePoolClients, latestBlockSearched, true)),
+    };
```
Comment (pxrl) [inline src/dataworker/Dataworker.ts:362]: @nicholaspai Making copies of the proposeRootBundle() function output is a pre-existing pattern in the Datworker. Do you know why we've done this historically?

File: src/dataworker/Dataworker.ts [hunk 2]
```diff
@@ -1112,37 +820,14 @@ export class Dataworker {
...
-        ..._rootBundleData,
-      };
-    }
+    const _rootBundleData = await this._proposeRootBundle(
+      blockRangesImpliedByBundleEndBlocks,
+      spokePoolClients,
+      rootBundle.proposalBlockNumber,
+      true
+    );
+    const rootBundleData = { ..._rootBundleData };
```
Comment (pxrl) [inline src/dataworker/Dataworker.ts:829]: Can possibly be simplified subject to https://github.com/across-protocol/relayer-v2/pull/1161/files#r1469027107.

File: src/dataworker/Dataworker.ts [hunk 3]
```diff
@@ -1332,38 +1016,13 @@ export class Dataworker {
...
-          } else {
-            const _rootBundleData = await this.UBA_proposeRootBundle(blockNumberRanges, ubaClient, spokePoolClients);
-            rootBundleData = {
-              ..._rootBundleData,
-            };
-          }
+            spokePoolClients,
+            matchingRootBundle.blockNumber
+          );
+          const rootBundleData = { ..._rootBundleData };
```
Comment (pxrl) [inline src/dataworker/Dataworker.ts:1024]: Can possibly be simplified subject to https://github.com/across-protocol/relayer-v2/pull/1161/files#r1469027107.

File: src/dataworker/Dataworker.ts [hunk 4]
```diff
@@ -2015,38 +1671,12 @@ export class Dataworker {
...
-        } else {
-          const _rootBundleData = await this.UBA_proposeRootBundle(blockNumberRanges, ubaClient, spokePoolClients);
-          rootBundleData = {
-            ..._rootBundleData,
-          };
-        }
+          spokePoolClients,
+          matchingRootBundle.blockNumber
+        );
+        const rootBundleData = { ..._rootBundleData };
```
Comment (pxrl) [inline src/dataworker/Dataworker.ts:1679]: Can possibly be simplified subject to https://github.com/across-protocol/relayer-v2/pull/1161/files#r1469027107.

File: src/dataworker/Dataworker.ts (hunk 2 above)
Comment (nicholaspai) [reply src/dataworker/Dataworker.ts:829]: I think the reason is that I wanted to maintain a variable `rootBundleData` that would be set either to the output of `Legacy_proposeRootBundle` or `UBA_proposeRootBundle`. I *don't think* the deep-copy of the object is an important detail and we can simplify these to `const rootBundelData = await ...`

**Review by nicholaspai — APPROVED** [review]
This will be a nice weight off our shoulders

**Review by james-a-morris — APPROVED** [review]
+1

File: src/dataworker/Dataworker.ts (hunk 4 above)
Comment (pxrl) [reply src/dataworker/Dataworker.ts]: https://github.com/across-protocol/relayer-v2/pull/1161/commits/592cf0f488025136e1f904043c7d2161b41da403

File: src/dataworker/Dataworker.ts (hunk 3 above)
Comment (pxrl) [reply src/dataworker/Dataworker.ts]: https://github.com/across-protocol/relayer-v2/pull/1161/commits/592cf0f488025136e1f904043c7d2161b41da403

File: src/dataworker/Dataworker.ts (hunk 2 above)
Comment (pxrl) [reply src/dataworker/Dataworker.ts]: @nicholaspai Updated here: https://github.com/across-protocol/relayer-v2/pull/1161/commits/592cf0f488025136e1f904043c7d2161b41da403

OK?

File: src/dataworker/Dataworker.ts (hunk 1 above)
Comment (pxrl) [reply src/dataworker/Dataworker.ts]: Updated: https://github.com/across-protocol/relayer-v2/pull/1161/commits/592cf0f488025136e1f904043c7d2161b41da403

File: src/dataworker/Dataworker.ts (hunk 2 above)
Comment (pxrl) [reply src/dataworker/Dataworker.ts]: I validated almost 2 weeks of root bundles with this change. Each one checked out OK, so I'll merge as-is.

**Pull request merged**